win32/Dockerfile

```diff
@@ -4,7 +4,7 @@ VOLUME /source
 
 # Installs dependencies once and for all
 RUN dnf -y update --refresh && \
-    dnf -y install cmake git fpc gettext glib2-devel itstool libxslt make yelp-tools zip && \
+    dnf -y install cmake git fpc gettext glib2-devel itstool libxslt libsoup3-devel make webkit2gtk4.1-devel yelp-tools zip && \
```

Comment:
Please note that this is running on Fedora 30. You aren't going to get the latest version of packages on it. For example, I can guarantee you that webkit2gtk4 is not going to be in there, as Fedora 30 reached end of life in 2020. If Xiphos' Windows build cannot be built against the antique versions of the packages found in Fedora 30, then we will need to find another way to do this. The Windows build has been stuck in limbo on very old versions of libraries for a long time because of these limitations. Specifically, though, you will need to install the mingw32/64 versions of these two libraries in order to build.

Reply:
because there have never been (successful) GTK3 or later packages for Windows, we remain stuck using GTK2. the code has been significantly #ifdef'd for this purpose for many years. (the original Windows port was in late 2009, when GTK2 was in its heyday. we have never moved off of it.)

yes, we know it's ancient and crufty and in some odd sense a security problem, but the use case of xiphos doesn't lend itself to any serious security abuse. it's what we've got to work with, so we build with it.

Reply:
...so that means, for Windows purposes, we have to return package qualifiers to the older versions. yes, webkit2gtk4.1 isn't available, and webkitgtk (original for GTK2) requires libsoup2.

Reply:
Correct. I was saying this for @Phyrik's edification.

There isn't a libsoup3 build (in Fedora) for MinGW. And the latest webkitgtk seems to be mingw32-webkitgtk3, which is only in very old releases of Fedora hence why we pinned this effort to Fedora 30. No webkit2 at all in the MinGW builds as that was never supported upstream. So the dependencies for Windows will need to specifically work with a version of libsoup < 3 and with the older version of webkit.

Reply:
I have pushed up a corrected version of the Dockerfile that does the Windows builds. You can find it on the tip of my branch with the same name as the PR branch here. It doesn't fix the actual build issues, as that is dependent on changes to the C code itself. However, if you follow the two steps in the CI script, you can do the development locally to get it into a good state with those deps.

`docker build -t xiphos win32` from the root of the Xiphos repository will create a local cached version of the build with all the dependencies intsalled. I use `podman` instead of `docker`, and there is no reason you can't, as well, if you prefer. It is the generally better alternative, but is a drop in replacement for you.

After that `docker run -i --rm -v ".:/source" xiphos -win32` and `docker run -i --rm -v ".:/source" xiphos -win64` will test building your local copy against Win32 and Win64 targets (we might want to drop the 32 bit target, eventually, because literally no one has 32 bit architecture these days on Windows). You can use these two commands to iterate on the changes needed to build with libsoup2 and webkitgtk2.

Reply:
I'm also not entirely sure you're correct about the GTK still being stuck on 2. I'm pretty sure we have been using GTK3 for the Windows build for quite some time. Additionally [GTK 4 supports Windows](https://docs.gtk.org/gtk4/windows.html). The only thing that prevents Xiphos from using a unified GTK 4 code base across both environments is use of WebKitGTK which dropped all Windows support some time ago and will likely never bring it back without a paying customer to support the effort.

Moving off of WebKitGTK would be a monumental forklift for the project. But, moving forward, it might become necessary. To date, the only really viable options I've seen are the [webview](https://github.com/webview/webview) library and (chromiumembeddedframework)[https://github.com/chromiumembedded/cef]. The former is more lightweight (wrapping existing system libraries on the host) while the latter is more heavyweight (embedded Chromium's rendering engine into your application). Both are well established, integrate with GTK, are widely used, and are intentionally cross platform. WebKitGTK is intentionally not.

I know it is a long-standing question. But the further in time we go, the more it might become important. GTK3 will eventually fall out of support and will stop supporting newer versions of Windows. If we're actually using GTK2 on Windows that will happen even sooner. It has continued to work for Xiphos for a while. But the end of the line is coming for using antiquated toolkits, eventually. And WebKitGTK is our only boat anchor tying us to them.

Reply:
> And WebKitGTK is our only boat anchor

call me cynical if you must, but when WebKitGTK came along, it was going to the permanent savior of all things HTML-rendered (because of course we _had_ to get rid of xulrunner) and would never be supplanted. so it became the 5th rendering engine known to gnomesword/xiphos.

now we have to look forward to a 6th.

_o frabjious joy._

Reply:
while i'm at it... in my current windows xiphos 4.2.1.4, installed under wine:
```
$ find -name *webkit*
./bin/libwebkitgtk-1.0-0.dll
./share/webkitgtk-1.0
```
is "1.0" actually gtk3?

Reply:
it says "WebKitGTK-2.0" right there on the package.
```
$ strings bin/libwebkitgtk-1.0-0.dll | c++filt | grep -i gtk.[123] | egrep -v _imp_gtk_ | sort | uniq
_head_libjavascriptcoregtk_1_0_0_dll
libjavascriptcoregtk-1.0-0.dll
libjavascriptcoregtk_1_0_0_dll_iname
libwebkitgtk-1.0-0.dll
libwebkitgtk-1.0-0.dll.debug
webkitgtk-1.0
WebKitGTK-2.0
```

Reply:
That 1.0 refers to the WebKit API that's used, not the GTK version. The current release is 6.0. The naming scheme jumped around a bit at first while they realized they needed to indicate three different versions: the WebKitGTK version, the WebKit API version, and the GTK version it's bound to. 

And I share your frustration with the moving target of engines. I think it was silly for WebKitGTK to drop support and I think it's a shame that there is no way to use Gecko outside of Firefox itself. But I lack the skills and time to maintain a rendering engine, so I'm left to the whims of those who do, sadly.